### PR TITLE
feat: Use `Fragmentizer` from `imap-next`

### LINKED ISSUE #536: refactor: Use `Fragmentizer` from imap-next

### PR DESCRIPTION
**Pull request by duesee**
Closes #536. This is only a move (with minor required changes to imports.)

### TODO

* ~Move fuzz-target~ Nevermind. Not pushed to imap-next so far.

### CODE REVIEW

File: imap-codec/src/lib.rs [hunk 1]
```diff
@@ -136,5 +137,6 @@ pub mod fuzz {
 }
 
 pub use codec::*;
+pub use fragmentizer::*;
```
Comment (jakoschiko) [inline imap-codec/src/lib.rs:140]: Are you sure about this?

- `fragmentizer` contains types with really generic names
- `fragmentizer` is only relevant for decoding

Alternatives that I like:

- Public module `imap_codec::fragmentizer`
- Public module `imap_codec::decode::fragmentizer`
- One of the options above and re-export only `imap_codec::Fragmentizer`

**Comment by jakoschiko** [comment]
Should we document the `Fragmentizer` somewhere?

- `README.md`?
- `lib.rs`?

File: imap-codec/src/lib.rs (hunk 1 above)
Comment (duesee) [reply imap-codec/src/lib.rs:140]: Let me think about it more. Didn't gave it enough thought.

**Comment by duesee** [comment]
> Should we document the `Fragmentizer` somewhere?
> 
>     * `README.md`?
> 
>     * `lib.rs`?

Yes. But let's do it in a later step.

**Comment by duesee** [comment]
I like the `fragmentizer` module idea. I'm not sure we already want to make it too decode-specific because I feel we may want to reuse `FragmentInfo` for encoding. I'm also a fan to not expose types twice. Thus, I think it's okay for now to have `Fragmentizer` available only via `fragmentizer::Fragmentizer` and not on the top-level, too. What do you think?

The idea of `parse_tag` was to allow users to obtain a tag on discarded data, right? So we want to expose it this way?

File: imap-codec/src/fragmentizer.rs [hunk 2]
```diff
@@ -0,0 +1,1401 @@
...
+            let parsed_literal = FragmentInfo::Literal {
+                start: self.start,
+                end: self.end,
+            };
+            (parsed_byte_count, Some(parsed_literal))
+        }
+    }
+}
+
+pub fn parse_tag(message_bytes: &[u8]) -> Option<Tag> {
```
Comment (jakoschiko) [inline imap-codec/src/fragmentizer.rs:530]: I made this public by accident.
```suggestion
fn parse_tag(message_bytes: &[u8]) -> Option<Tag> {
```

**Pull request merged**